text/mavlink2_1.md

```diff
@@ -0,0 +1,62 @@
+  * Start date: 2022-10-12
+  * Contributors: Andrew Tridgell
+
+# Summary
+
+Rough proposal for MAVLink 2.1
+
+# Motivation
+
+Key features wanted:
+
+ * support for more than 256 system IDs
+ * support for checking CRCs in routers that don't have all xml
+ * better target_system/target_component support
+ * maximum compatibility with existing mavlink2 systems
+
+# Detailed Design
+
+The design takes advantage of the incompat_flags and compat_flags
+fields in the MAVLink2 header.
+
+New incompat_flags flags would be:
+
+ * MAVLINK_IFLAG_SYSID16 : system ID in header is 16 bits wide
+ * MAVLINK_IFLAG_TARGETTED : header has target_system and target_component appended
+ * MAVLINK_IFLAG_SPLIT_CRC : 16 bit crc is as described below
+
+New compat_flags flags would be:
+
+ * MAVLINK_FLAG_21_CAPABLE : sender is capable of handling MAVLink 2.1
+
+Note that for the SYSID16 and TARGETTED flags the sender should only
+use the flag if is it needed. So if the systemID in the message is
+<=255 then the flag should not be set and the existing 8 bit field
+should be used. For the TARGETTED bit this should only be used if the
+target systemID is >255 (needing 16 bits) or if the message that needs
+targetting doesn't have a target_system or target_component field.
+
+Following these rules maximises compatibility with existing MAVLink2
+capable systems.
+
+Note that when SYSID16 flag is set then the target_system would be 16
+bits when TARGETTED is set.
+
+For messages with existing target_system/target_component fields, if
+the TARGETTED bit is set then the existing fields are ignored by the
+recipient. The sender should set the old fields to zero.
```

Comment:
Can we strip the ids out out of the payload in this case by default in the generator?

Reply:
no, I don't think so, I think that would complicate the parsers too much

Reply:
Sending the data twice, forever, seems like something we should try avoid. Is there a future where we can remove the ids from the XML then?

Jumping back a little, currently a message has the target ids in the payload, and knows to add them from the message definition. Other fields like sender ids are always present.

How will the library know that it should add target ids to the header? You might be assuming they will always be added for systems after 256 or you might be assuming these will still be added conditionally to the header based on the message definition.

Reply:
they are added to the header when the value is >255 and the incompat flag is set
the receiver uses the value from the header when the incompat flag is set